code/obj/item/gun/kinetic.dm

```diff
@@ -623,23 +623,67 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
 	name = "Riot Shotgun"
 	desc = "A police-issue shotgun meant for suppressing riots."
 	icon = 'icons/obj/48x32.dmi'
-	icon_state = "shotty"
+	icon_state = "shotty-empty"
 	item_state = "shotty"
 	force = 15.0
 	contraband = 5
 	caliber = 0.72
 	max_ammo_capacity = 8
-	auto_eject = 1
+	auto_eject = 0
 	can_dual_wield = 0
 	two_handed = 1
 	has_empty_state = 1
 	gildable = 1
+	var/racked_slide = FALSE
+
 
 	New()
 		ammo = new/obj/item/ammo/bullets/abg
 		set_current_projectile(new/datum/projectile/bullet/abg)
 		..()
 
+	update_icon()
+		. = ..()
+		src.icon_state = "shotty" + (gilded ? "-golden" : "") + (racked_slide ? "" : "-empty" )
+
+	canshoot()
+		if (src.racked_slide)
+			return ..()
+		else
+			boutput(usr, "<span class='notice'>You need to rack the slide before you can fire!</span>") //Pain
+			return FALSE
+
+	shoot(var/target,var/start ,var/mob/user)
+		..()
+		src.racked_slide = FALSE
+		if (src.ammo.amount_left == 0) // change icon_state to empty if 0 shells left
+			src.update_icon()
+		src.casings_to_eject = 0
+
+	attack_self(mob/user as mob)
+		..()
+		if (!src.racked_slide) //Are we racked?
+			if (src.ammo.amount_left == 0)
+				boutput(user, "<span class ='notice'>You are out of shells!</span>")
+				update_icon()
+			else
+				src.racked_slide = TRUE
+				if (src.icon_state == "shotty[src.gilded ? "-golden" : ""]") //"animated" racking
+					src.icon_state = "shotty[src.gilded ? "-golden-empty" : "-empty"]" // having update_icon() here breaks
+					animate(src, time = 0.2 SECONDS)
+					animate(icon_state = "shotty[gilded ? "-golden" : ""]")
+				else
+					update_icon() // Slide already open? Just close the slide
+				boutput(user, "<span class='notice'>You rack the slide of the shotgun!</span>")
+				playsound(user.loc, "sound/weapons/shotgunpump.ogg", 50, 1)
+				if (src.ammo.amount_left < 8) // Do not eject shells if you're racking a full "clip"
```

Comment:
If you fire a shot, reload the shotgun, and rack the shotgun, this code will not produce a casing

Reply:
This is intentional. The shotgun starts unracked with the slide open, and the initial rack is merely loading a shell into the chamber, and not ejecting one.

Reply:
If I fire a shot, I would expect there to be a casing in the gun, which would come out the next time it is racked. If you reload the gun before racking it, no casing will be produced upon racking the gun.

Reply:
tested in game and now i see what you mean. ill work on a fix

Reply:
I believe I fixed it, but its honestly not pretty and I wish I could've done it better, but I don't think it would be worth the time investment.